src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -85,6 +114,12 @@ public Double averageSumOdd(List<Integer> numbers) {
      * Задача: вивести все имена кошек в которых хозяева это девушки старше 18 лет</p>
      **/
     public List<String> getCatsNames(List<People> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getAge() > femaleAge
+                        && p.getSex() == People.Sex.WOMEN
+                        && !p.getCatList().isEmpty())
+                .flatMap(p -> p.getCatList().stream())
```

Comment:
@TatianaPirch Can you do in this way?
```suggestion
                .map(p -> p.getCatList())
                .flatMap(List::stream)
```